irohad/consensus/yac/impl/yac.cpp

```diff
@@ -24,336 +24,349 @@ auto &getRound(const std::vector<iroha::consensus::yac::VoteMessage> &state) {
   return state.at(0).hash.vote_round;
 }
 
-namespace iroha {
-  namespace consensus {
-    namespace yac {
-
-      std::shared_ptr<Yac> Yac::create(
-          YacVoteStorage vote_storage,
-          std::shared_ptr<YacNetwork> network,
-          std::shared_ptr<YacCryptoProvider> crypto,
-          std::shared_ptr<Timer> timer,
-          ClusterOrdering order,
-          Round round,
-          rxcpp::observe_on_one_worker worker,
-          logger::LoggerPtr log) {
-        return std::make_shared<Yac>(vote_storage,
-                                     network,
-                                     crypto,
-                                     timer,
-                                     order,
-                                     round,
-                                     worker,
-                                     std::move(log));
+namespace iroha::consensus::yac {
+
+  std::shared_ptr<Yac> Yac::create(YacVoteStorage vote_storage,
+                                   std::shared_ptr<YacNetwork> network,
+                                   std::shared_ptr<YacCryptoProvider> crypto,
+                                   std::shared_ptr<Timer> timer,
+                                   ClusterOrdering order,
+                                   Round round,
+                                   rxcpp::observe_on_one_worker worker,
+                                   logger::LoggerPtr log) {
+    return std::make_shared<Yac>(vote_storage,
+                                 network,
+                                 crypto,
+                                 timer,
+                                 order,
+                                 round,
+                                 worker,
+                                 std::move(log));
+  }
+
+  Yac::Yac(YacVoteStorage vote_storage,
+           std::shared_ptr<YacNetwork> network,
+           std::shared_ptr<YacCryptoProvider> crypto,
+           std::shared_ptr<Timer> timer,
+           ClusterOrdering order,
+           Round round,
+           rxcpp::observe_on_one_worker worker,
+           logger::LoggerPtr log)
+      : log_(std::move(log)),
+        cluster_order_(order),
+        round_(round),
+        worker_(worker),
+        notifier_(worker_, notifier_lifetime_),
+        vote_storage_(std::move(vote_storage)),
+        network_(std::move(network)),
+        crypto_(std::move(crypto)),
+        timer_(std::move(timer)) {}
+
+  Yac::~Yac() {
+    notifier_lifetime_.unsubscribe();
+  }
+
+  void Yac::stop() {
+    network_->stop();
+  }
+
+  // ------|Hash gate|------
+
+  void Yac::vote(YacHash hash,
+                 ClusterOrdering order,
+                 boost::optional<ClusterOrdering> alternative_order) {
+    log_->info(
+        "Order for voting: [{}]",
+        boost::algorithm::join(
+            order.getPeers() | boost::adaptors::transformed([](const auto &p) {
```

Comment:
Message length could be toooooo long if number of peers in network big. Let's think about compact log messages.